rotkehlchen/exchanges/iconomi.py

```diff
@@ -224,29 +229,25 @@ def query_balances(self, **kwargs: Any) -> ExchangeQueryBalances:
             try:
                 asset = asset_from_iconomi(ticker)
 
-                # There seems to be a bug in the ICONOMI API regarding balance_info['value'].
-                # The value is supposed to be in USD, but is actually returned
-                # in EUR. So let's use the Inquirer for now.
                 try:
-                    usd_price = Inquirer().find_usd_price(asset=asset)
-                except RemoteError as e:
-                    self.msg_aggregator.add_error(
-                        f'Error processing ICONOMI balance entry due to inability to '
-                        f'query USD price: {str(e)}. Skipping balance entry',
+                    usd_value = deserialize_fval(balance_info['value'], 'usd_value', 'iconomi')
+                except (DeserializationError, KeyError) as e:
+                    self.msg_aggregator.add_warning(
```

Comment:
For KeyError don't catch it like that.

Edit the message like we do for example here: https://github.com/rotki/rotki/blob/660d995bed399bd3d375acf8bc322aedd4ef0a9f/rotkehlchen/exchanges/coinbasepro.py#L380-L383

The reason is that `str(e)` for a `KeyError` is going to be just the keyname.